src/agents/run.py

```
@@ -208,7 +208,20 @@ async def run(
                                 data={"max_turns": max_turns},
                             ),
                         )
-                        raise MaxTurnsExceeded(f"Max turns ({max_turns}) exceeded")
+                        run_error_details = RunErrorDetails(
+                            input=original_input,
+                            new_items=generated_items,
+                            raw_responses=model_responses,
+                            final_output=None,
+                            input_guardrail_results=input_guardrail_results,
+                            output_guardrail_results=[],
+                            context_wrapper=context_wrapper,
+                            _last_agent=current_agent
+                        )
```

it might make more sense to create this in the outermost `try-catch`. You'll catch all exceptions there, so you can capture and re-raise